src/search/parser_tests.h

```diff
@@ -0,0 +1,210 @@
+        : SearchEngine(opts),
+          searches() {
+        std::shared_ptr<parser::EnrichedASTNode> searches_AST =
+            opts.get<std::shared_ptr<parser::EnrichedASTNode>>("searches");
+        std::shared_ptr<parser::EnrichedListNode> searches_AST_list =
+                std::dynamic_pointer_cast<parser::EnrichedListNode>(searches_AST);
+        for (const auto & element : searches_AST_list->get_elements()) {
+            searches.push_back(element->clone());
+        }
+    }
+    virtual ~IteratedSearch() = default;
+    void dump(std::string indent) override {
+        std::cout << indent << "IteratedSearch with " << std::endl;
+        indent += "    ";
+        std::cout << indent << "searches = " << std::endl;
+        for (const auto& search_AST : searches) {
+            parser::Any search_any = search_AST->construct();
+            std::shared_ptr<SearchEngine> search = parser::any_cast<std::shared_ptr<SearchEngine>>(search_any);
+            search->dump(indent + "    ");
+        }
+    }
+};
+
+class IteratedSearchPlugin : public parser::Plugin<SearchEngine, IteratedSearch> {
+public:
+    IteratedSearchPlugin()
+        : parser::Plugin<SearchEngine, IteratedSearch>("iterated", "Iterated Search") {;
+        add_option<std::vector<std::shared_ptr<SearchEngine>>>(
+            "searches",
+            "use preferred operators of these evaluators", "[]", parser::Bounds::unlimited(), true);
+    }
+};
+
+class BlindHeuristicPlugin : public parser::Plugin<Evaluator, BlindHeuristic> {
+public:
+    BlindHeuristicPlugin()
+        : parser::Plugin<Evaluator, BlindHeuristic>("blind", "Blind Heuristic") {
+        document_synopsis("Returns cost of cheapest action for "
+                          "non-goal states, "
+                          "0 for goal states");
+        document_language_support("action costs", "supported");
+        document_language_support("conditional effects", "supported");
+        document_language_support("axioms", "supported");
+        document_property("admissible", "yes");
+        document_property("consistent", "yes");
+        document_property("safe", "yes");
+        document_property("preferred operators", "no");
+
+        add_option<int>("repeat", "how often to check for the goal", "2K");
+        add_option<double>("time_limit", "when to stop", "2e4");
+        add_list_option<std::vector<int>>("patterns", "...", "[[1,2,3], []]");
+
+        std::vector<std::string> lp_solvers;
+        std::vector<std::string> lp_solvers_doc;
+        lp_solvers.push_back("CLP");
+        lp_solvers_doc.push_back("default LP solver shipped with the COIN library");
+        lp_solvers.push_back("CPLEX");
+        lp_solvers_doc.push_back("commercial solver by IBM");
+        lp_solvers.push_back("GUROBI");
+        lp_solvers_doc.push_back("commercial solver");
+        lp_solvers.push_back("SOPLEX");
+        lp_solvers_doc.push_back("open source solver by ZIB");
```

Comment:
We actually have a nicer way of doing this, which is now done inside the `add_enum_option` method. It would look like this:

```C++
EnumInfo lp_solver_enum;
lp_solver_enum.add_option("CLP", "default LP solver shipped with the COIN library");
lp_solver_enum.add_option("CPLEX", "commercial solver by IBM");
lp_solver_enum.add_option("GUROBI", "commercial solver");
lp_solver_enum.add_option("SOPLEX", "open source solver by ZIB");
```

We didn't use it outside of `add_enum_option` yet to reduce the diff to the old code a bit but maybe this is unnecessary since we have to touch that part of the code anyway and it would be better to directly bring it into the form we want it to have. Should I change it?

Reply:
There are still parallel data structures inside `EnumInfo` but that is then an implementation detail of that class.

The thing that is a bit unpleasant about this whole part of the code is that theoretically, we can have the same enum with different documentation in different plugins. Our use case for this was that for example a log level of `INFO`, `DEBUG`, etc. could be documented in the classes where it is used by documenting what exactly is logged with this level. In any case, if the same enum is used in different places, we pass documentation for it each time and cannot easily guarantee that it is the same. The way we currently handle this is that there are two structures related to the enum. `EnumInfo` is used inside the plugin to describe and document the options of a particular enum argument. But then the registry stores `EnumValues` (an alias for `unordered_map<type_index, vector<std::string>>`)  that map the type index of an enum to the strings of all its values. This is used for parsing and at the time we create the registry, we check that the values we use for an enum class at all places were it occurs as an argument are consistent.